src/sql-server-util/src/cdc.rs

```diff
@@ -95,6 +96,30 @@ impl<'a> CdcStream<'a> {
             crate::inspect::get_tables_for_capture_instance(self.client, instances).await?;
         tracing::info!(?tables, "got table for capture instance");
 
+        // Before we start a transaction, ensure SQL Server is reporting an LSN.
+        //
+        // If CDC was very recently enabled (e.g. < 30 seconds ago) then it will
+        // report NULL for an LSN. Before we start a transaction, which prevents
+        // the LSN from progressing, we want to wait a bit for SQL Server to
+        // become ready.
+        let (_client, lsn_result) = mz_ore::retry::Retry::default()
+            .max_duration(std::time::Duration::from_secs(10))
```

Comment:
Does this give 10 seconds to the query to come back? The default might be too tight, and definitely worth a dyncfg

Reply:
It does, this should only really matter when CDC is first initialized for the entire DB, but is something I ran into in tests. I made this configurable via a dyncfg and updated the max wait to 30 seconds